docs/design/001-interchain-queries.md

```diff
@@ -0,0 +1,129 @@
+# Design Doc 001: Interchain Queries
+
+## Status
+
+Draft
+
+## Abstract
+
+In order for the Ojo network to effectively provide services to other chains, we need a method of transport at Layer 1. We must develop two mechanisms:
+
+*Client* Module - accepts validator responses to requests made by the *Client*; requires a *Proof* for information to be accepted.
+*Relayer* - allows relayers to move information between Ojo and the *Client*, and is responsible for submitting proofs.
+
+The Client chain will request data, which the relayer will then pick up, query Ojo for the requested data, and then respond back to the original chain.
+
+## Context
+
+There are two main existing solutions :
+
+### [Quicksilver's Interchain Queries](https://github.com/ingenuity-build/quicksilver/tree/main/x/interchainquery/keeper)
+
+Quicksilver requires accurate information about other chains in order to utilize interchain accounts for their remote liquid staking solution.
+
+This is a KV store solution which requires the *Client* and *Relayer* infrastructure, but does not require a *Host* module. Quicksilver also requires validators to prove that data is accurate before accepting it on-chain.
+
+### [Strangelove's IBC Module](https://github.com/strangelove-ventures/ibc-go/tree/feature/icq_implementation/modules/apps/icq)
+
+The ICQ IBC module fully leverages IBC standards, and requires the *Client*, *Relayer*, and *Host* pieces of infrastructure. It also requires a proof to the *Client* that the information from the *Host* is accurate.
+
+There are a few issues with bringing both of these solutions in-house:
+
+1. Ease of API - The quicksilver & strangelove implementations are for agnostic data, and in order to consume relayed packets, the *Client* chain would have to parse this. This should be a part of the *Client* module.
+2. Economic Factor - Neither implementation allows for relayers to be paid for their services; in fact, in both implementation if the gas fee of a chain is above 0, it *costs* coins to run a relayer between both chains. The economic model in this case will be dealt with in a future design doc.
+
+## Specification
+### Client
+
+#### Msgs
+
+- `FulfillDataRequest(data, proof)` - Fulfills the data request submitted on the `Host` module. Requires `Proof` that the information from the `Host` actually belongs to the host using an [ABCI RequestQuery.](https://github.com/strangelove-ventures/ibc-go/tree/feature/icq_implementation/modules/apps/icq#abci-query).
```

Comment:
We're still referencing a host module here

Reply:
good catch, I just flipped this I think 😅 23f40b0a493222fa337f31be3b5414ad0d843696